packages/next/src/build/webpack/loaders/next-flight-loader/index.ts

```diff
@@ -97,21 +97,24 @@ export default function transformSource(
         return
       }
 
+      // `proxy` is the module proxy that we treat the module as a client boundary.
+      // For ESM, we access the property of the module proxy directly for each export.
+      // This is bit hacky that treating using a CJS like module proxy for ESM's exports,
+      // but this will avoid creating nested proxies for each export. It will be improved in the future.
       let esmSource = `\
 import { createProxy } from "${MODULE_PROXY_PATH}"
+
+const proxy = createProxy(String.raw\`${resourceKey}\`)
 `
       let cnt = 0
       for (const ref of clientRefs) {
         if (ref === '') {
-          esmSource += `\nexports[''] = createProxy(String.raw\`${resourceKey}#\`);`
+          esmSource += `exports[''] = proxy['']\n`
         } else if (ref === 'default') {
-          esmSource += `\
-export default createProxy(String.raw\`${resourceKey}#default\`);
-`
+          esmSource += `export default createProxy(String.raw\`${resourceKey}#default\`)\n`
```

Comment:
why isn't this `export default proxy`?

Reply:
@gnoff I made it work to use proxy.default in #66990